lib/ansible/modules/cloud/vmware/vmware_guest_file_operation.py

```diff
@@ -0,0 +1,379 @@
+    file_manager = content.guestOperationsManager.fileManager
+    if operation == "create":
+        try:
+            file_manager.MakeDirectoryInGuest(vm=vm, auth=creds, directoryPath=path,
+                                              createParentDirectories=recurse)
+        except vim.fault.FileAlreadyExists:
+            module.fail_json(msg="Directory %s already exist" % path)
+        except vim.fault.GuestPermissionDenied:
+            module.fail_json(msg="Permission denied for path %s" % path)
+        except vim.fault.InvalidGuestLogin:
+            module.fail_json(msg="Invalid guest login for user %s" % vm_username)
+        # other exceptions
+        except Exception as e:
+            module.fail_json(msg="Failed to Create directory into Vm VMware exception:%s" % e)
+
+    if operation == "delete":
+        try:
+            file_manager.DeleteDirectoryInGuest(vm=vm, auth=creds, directoryPath=path,
+                                                recursive=recurse)
+        except vim.fault.FileFault as e:
+            module.fail_json(msg="FileFault:%s" % e.msg)
+        except vim.fault.GuestPermissionDenied:
+            module.fail_json(msg="Permission denied for path %s" % path)
+        except vim.fault.InvalidGuestLogin:
+            module.fail_json(msg="Invalid guest login for user %s" % vm_username)
+        # other exceptions
+        except Exception as e:
+            module.fail_json(changed=False, msg="Failed to Delete directory into Vm VMware exception:%s" % e)
+
+    return True
+
+
+def fetch(module, content, vm):
+    vm_username = module.params['vm_username']
+    vm_password = module.params['vm_password']
+
+    if "src" not in module.params["fetch"]:
+        module.fail_json(msg="fetch.src is mandatory")
+    if "dest" not in module.params["fetch"]:
+        module.fail_json(msg="fetch.dest is mandatory")
+
+    dest = module.params["fetch"]['dest']
+    src = module.params['fetch']['src']
+
+    creds = vim.vm.guest.NamePasswordAuthentication(username=vm_username, password=vm_password)
+    file_manager = content.guestOperationsManager.fileManager
+
+    try:
+        fileTransferInfo = file_manager.InitiateFileTransferFromGuest(vm=vm, auth=creds,
+                                                                      guestFilePath=src)
+    except vim.fault.FileFault as e:
+        module.fail_json(msg="FileFault:%s" % e.msg)
+    except vim.fault.GuestPermissionDenied:
+        module.fail_json(msg="Permission denied to fetch file %s" % src)
+    except vim.fault.InvalidGuestLogin:
+        module.fail_json(msg="Invalid guest login for user %s" % vm_username)
+    # other exceptions
+    except Exception as e:
+        module.fail_json(changed=False, msg="Failed to Fetch file from Vm VMware exception:%s" % e)
+
+    url = fileTransferInfo.url
+    resp, info = urls.fetch_url(module, url, method="GET")
+    with open(dest, "wb") as local_file:
+        local_file.write(resp.read())
```

Comment:
Do we need to do any check on this? Can this fail due to permission issues, or due to the file already existing?

I'm fine with this always overwriting the destination file, and coming up `changed=True` every time, i just wonder if there is a risk of exceptions (pretty sure both `fetch_url`, `open` and `write` can throw a bunch of exceptions.)